### PR TITLE
Add support for get parameters

### PR DESCRIPTION
**Pull request by mariane-bekker**

### CODE REVIEW

File: demo/index.js [hunk 1]
```diff
@@ -91,6 +91,10 @@ api.route('/users/:userId/photos', photoCollectionSchema, {
     // The API router will figure out which callbacks are necessary to satisfy the incoming request.
     get: function(request) {
         var userId = request.getParam('userId');
+
+        //Get Parameter
+        var offset = request.getQuery('offset');
```
Comment (djvirgen) [inline demo/index.js]: Where is this used?

File: lib/Request.js [hunk 2]
```diff
@@ -88,6 +89,24 @@ Request.prototype.getParam = function(param) {
...
+
+Request.prototype.setQuery = function(query, value) {
+    this._url.query[query] = value;
+    return this;
+};
+
+Request.prototype.getQuery = function(query) {
+    return this._url.query.hasOwnProperty(query) ? this._url.query[query] : undefined;
+};
+
```
Comment (kelfman) [inline lib/Request.js:109]: Just a thought, have you considered using getters and setters here?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects#Defining_getters_and_setters

File: lib/Connection.js [hunk 3]
```diff
@@ -29,6 +29,13 @@ function Connection(router, req, res) {
         if (options) {
             if (options.props) request.setProps(options.props);
             if (options.resource) request.setResource(options.resource);
+
+            if (options.query) {
+                for (var key in options.query) {
```
Comment (eviechan) [inline lib/Connection.js:34]: Query objects should be passed in with `?abc=def&foo=bar`, etc., and not as map of key/value pairs.

Comment (eviechan) [inline lib/Connection.js:34]: You'll have to do some splitting with `&` and then `=` to get maps of key/value pairs to work with.

File: test/lib/Router_test.js [hunk 4]
```diff
@@ -61,6 +61,55 @@ describe('API Router', function() {
...
+                }.bind(this))
+                .finally(done);
+            });
+
+            it('resolves with object from callback with route having a parameter in end of the url', function(done) {
+                this.router.route('/foo/test/:fooId', this.fooSchema, {
+                    get: this.getParamsFoo
+                });
+                this.connection.get('/foo/test/123', {
+                  query: { param1: 1, param2: 'test' }
```
Comment (eviechan) [inline test/lib/Router_test.js:99]: I don't think query parameters should be passed in as such for a URL.

File: lib/Connection.js (hunk 3 above)
Comment (djvirgen) [inline lib/Connection.js:34]: No, this is correct. The interface here is `connection#get(resourceId[, options])`. In Monocle clients, that will be translated to an HTTP request with stringified query string parameters:

```
connection.get('/users', {
  query: {
    limit: 10,
    offset: 0
  }
})
```

Which would be converted to an HTTP GET:

```
GET /users?limit=10&offset=0
```

This is similar to how we handle `props`, which is passed in thusly:

```
connection.get('/users/123', {
  props: ['name', 'age']
})
```

Which is converted to an HTTP GET:

```
GET /users/123?props=name,age
```

The reason for this is to allow the resource ID to be used as a cache key and to make subsequent calls:

```
monocleClient.get('/me')
.then((user) => {
  user.name = 'Sally';
  // user.$resourceId may not be "/me", but the canonical path "/users/123"
  monocleClient.patch(user.$resourceId, {
   resource: user
  }).then((user) => {
    console.log('updated user!', user);
  });
});
```

File: test/lib/Router_test.js (hunk 4 above)
Comment (mariane-bekker) [inline test/lib/Router_test.js:99]: see hector's comment above

File: lib/Request.js (hunk 2 above)
Comment (mariane-bekker) [inline lib/Request.js:109]: this is how we currently do it for all properties.... if we do want to use getters and setters we will have to do it for the file... i can put in a ticket to investigate but i'd like to get this in for now so we can start building our end points requiring pagination

File: lib/Connection.js (hunk 3 above)
Comment (eviechan) [inline lib/Connection.js:34]: I see.  I had thought it was the way the URL would be passed in, as opposed to construction of URL with the connection object we get.  My bad.

File: lib/Request.js (hunk 2 above)
Comment (djvirgen) [inline lib/Request.js:109]: getters and setters could work for the `setQueries` and `getQueries` methods, but the interface is not compatible with `getQuery(name)` and `setQuery(name, value)` unless we promote `this._url.query` to a new object.

Either way, the public interface doesn't change, but this implementation probably requires the least amount of code.

**Comment by djvirgen** [comment]
:cherries:

**Pull request merged**